### PR TITLE
Recommend cluster scope for Metricbeat

### LINKED ISSUE #70853: Elasticsearch monitoring docs recommend installing Metricbeat on every node regardless of cluster config
Today the [instructions for configuring Metricbeat to monitor an Elasticsearch cluster](https://www.elastic.co/guide/en/elasticsearch/reference/current/configuring-metricbeat.html) recommend installing Metricbeat on every node. This is good advice in smaller clusters, but once a cluster has dedicated master-eligible nodes it's no longer what we should recommend. By default Metricbeat will collect most of its Elasticsearch metricsets from the elected master, but if the elected master is a dedicated master-eligible node then [we should not be sending requests to it](https://www.elastic.co/guide/en/elasticsearch/reference/current/high-availability-cluster-small-clusters.html#high-availability-cluster-design-three-plus-nodes):

> In larger clusters, we recommend you configure some of your nodes as dedicated master-eligible nodes and avoid sending any client requests to these dedicated nodes. Your cluster may become unstable if the master-eligible nodes are overwhelmed with unnecessary extra work that could be handled by one of the other nodes.

Instead in a larger cluster I think we should recommend running a single Metricbeat with `scope: cluster` and without the addresses of the master nodes.

Linking https://github.com/elastic/elasticsearch/issues/70435 since this would prevent the Metricbeat-per-node configuration from working properly in a cluster with dedicated master-eligible nodes.

### PR DESCRIPTION
**Pull request by DaveCTurner**
Today we mention Metricbeat's `scope` parameter but offer no guidance
about how it should be used. This commit adds guidance to use `scope:
cluster`, especially on clusters with dedicated master-eligible nodes.

Closes #70853

### CODE REVIEW

File: docs/reference/monitoring/configuring-metricbeat.asciidoc [hunk 1]
```diff
@@ -86,9 +94,12 @@ update the `hosts` setting. If you configured {es} to use encrypted
 communications, you must access it via HTTPS. For example, use a `hosts` setting
 like `https://localhost:9200`.
 <2> By default, `scope` is set to `node` and each entry in the `hosts` list
-indicates a distinct node in an {es} cluster. If you set `scope` to `cluster`,
-each entry in the `hosts` list indicates a single endpoint for a distinct {es}
-cluster (for example, a load-balancing proxy fronting the cluster).
+indicates a distinct node in an {es} cluster. If you set `scope` to `cluster`
+ithen each entry in the `hosts` list indicates a single endpoint for a distinct
```
Comment (thbkrkr) [inline docs/reference/monitoring/configuring-metricbeat.asciidoc:98]: Typo:
```suggestion
then each entry in the `hosts` list indicates a single endpoint for a distinct
```

Comment (DaveCTurner) [reply docs/reference/monitoring/configuring-metricbeat.asciidoc]: d'oh, thanks

**Comment by DaveCTurner** [comment]
@elasticmachine update branch

**Review by jakommo — APPROVED** [review]
LGTM!

File: docs/reference/monitoring/configuring-metricbeat.asciidoc [hunk 2]
```diff
@@ -42,8 +42,16 @@ view the cluster settings and `manage` cluster privileges to change them.
...
 
-. {metricbeat-ref}/metricbeat-installation-configuration.html[Install {metricbeat}] on each
-{es} node in the production cluster. Failure to install on each node may result in incomplete or missing results.
+. {metricbeat-ref}/metricbeat-installation-configuration.html[Install
+{metricbeat}]. Ideally install a single {metricbeat} instance configured with
+`scope: cluster` and configure `hosts` to point to an endpoint (e.g. a
+load-balancing proxy) which directs requests to the master-ineligible nodes in
+the cluster. If this is not possible, and the cluster does not use dedicated
+master-eligible nodes, then install one {metricbeat} instance for each {es}
+node in the production cluster and use the default `scope: node`. When
```
Comment (henningandersen) [inline docs/reference/monitoring/configuring-metricbeat.asciidoc:51]: I find the `and the cluster does not use dedicated master-eligible nodes` a bit too hard/prescriptive. I would prefer to add a separate sentence that this is not recommended with dedicated master-eligible nodes and requires beefier master nodes than when using `scope:cluster`.

**Review by henningandersen — COMMENTED** [review]
Left a comment, looking good otherwise.

File: docs/reference/monitoring/configuring-metricbeat.asciidoc (hunk 2 above)
Comment (DaveCTurner) [reply docs/reference/monitoring/configuring-metricbeat.asciidoc]: ok, see faa0147

**Review by henningandersen — APPROVED** [review]
LGTM.

**Comment by DaveCTurner** [comment]
Thanks all!

**Pull request merged**